src/r2mm/manager/SettingsDexieStore.ts

```diff
@@ -24,66 +24,47 @@ export default class SettingsDexieStore extends Dexie {
-                const globalEntry = result[result.length - 1];
-                const parsed = JSON.parse(globalEntry.settings);
-                if ((parsed as ManagerSettingsInterfaceGlobal_V2).version) {
-                    // Is modern (at least V2).
-                    return parsed;
-                } else {
-                    // Is legacy.
-                    const legacyToV2 = this.mapLegacyToV2(parsed, this.activeGame);
-                    await this.global.put({ settings: JSON.stringify(legacyToV2.global) });
-                    await this.gameSpecific.put({ settings: JSON.stringify(legacyToV2.gameSpecific) });
-                    return legacyToV2.global;
-                }
-            } else {
-                ManagerSettings.NEEDS_MIGRATION = true;
-                const obj = this.createNewSettingsInstance();
-                await this.global.put({ settings: JSON.stringify(obj.global) });
-                await this.gameSpecific.put({ settings: JSON.stringify(obj.gameSpecific) });
-                return obj.global;
-            }
-        });
+        const global = await this.global.get(1);
+
+        // Create the global settings row if it does not already exist.
+        if (global == undefined) {
+            const newSettings = this.createNewSettingsInstance();
+            await this.global.put({ settings: JSON.stringify(newSettings.global) });
+            return newSettings.global;
+        }
+
+        // Otherwise parse and return the settings field.
+        return JSON.parse(global.settings);
     }
 
     public async getLatestGameSpecific(): Promise<ManagerSettingsInterfaceGame_V2> {
-        return this.gameSpecific.toArray().then(async result => {
-            if (result.length > 0) {
-                const globalEntry = result[result.length - 1];
-                const parsed = JSON.parse(globalEntry.settings);
-                if ((parsed as ManagerSettingsInterfaceGame_V2).version === 2) {
-                    // Is modern (at least V2).
-                    return parsed;
-                } else {
-                    // Placeholder for future migration
-                    return;
-                }
-            } else {
-                const obj = this.createNewSettingsInstance();
-                await this.gameSpecific.put({ settings: JSON.stringify(obj.gameSpecific) });
-                return obj.gameSpecific;
-            }
-        });
+        const game = await this.games.get(this.activeGame.settingsIdentifier);
+
+        // If we don't have a settings row for this game, either put and return a new one or migrate.
+        if (game == undefined && !(await this.applyV2ToV3MigrationPlease())) {
+            let obj = this.createNewSettingsInstance();
+            await this.games.put({ identifier: this.activeGame.settingsIdentifier, settings: JSON.stringify(obj.gameSpecific) });
+
+            return obj.gameSpecific;
+        }
+
+        return JSON.parse(game!.settings);
```

Comment:
Doesn't using the non-null assertion here hide the situation where `game` is undefined and the V2ToV3 migration was successfully done?

Reply:
Resolved, thanks for catching that.